### DISCUSSION TITLE
MySQL connection randomly fails even though the healthcheck passed

### DISCUSSION THREAD

**Discussion by danielbachhuber** [community/community#52445] [Actions] [answered]
### Select Topic Area

Bug

### Body

We're running into an issue where our installation script randomly fails with this error:

```
> install-package-tests
Warning: arning] Using a password on the command line interface can be insecure.
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1:32768' (111)
/home/runner/work/extension-command/extension-command/vendor/wp-cli/wp-cli-tests/bin/install-package-tests: 81: [: Illegal number: 
Detected MySQL at version .
+ mysql -e CREATE DATABASE IF NOT EXISTS `wp_cli_test`; -h127.0.0.1 -P32768 --protocol=tcp -uroot -proot
Warning: arning] Using a password on the command line interface can be insecure.
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1:32768' (111)
Script install-package-tests handling the prepare-tests event returned with error code 1
Error: Process completed with exit code 1.
```

* Script in question: https://github.com/wp-cli/wp-cli-tests/blob/bf9299862874b51041c4fe8599336bd0ebf2393a/bin/install-package-tests
* Build where it fails: https://github.com/wp-cli/wp-cli-tests/actions/runs/4657659782/jobs/8242469998
* Workflow: https://github.com/wp-cli/.github/blob/143819e34752559c6ba453eb1ea389262b78838d/.github/workflows/reusable-testing.yml

Interestingly, the healthcheck passed at the beginning of the workflow:

<img width="1273" alt="image" src="https://user-images.githubusercontent.com/36432/230917026-ef688e4e-8a17-49f6-af6c-996b81a84eec.png">

Do you have any suggestions for how we might debug further?

Also tracking in https://github.com/wp-cli/wp-cli-tests/issues/160

**Comment by szepeviktor** [comment, marked as answer]
We were debugging this with John.
A second functional test (an SQL query) is the solution.
https://github.com/johnbillion/query-monitor/blob/5d62ca2344bb22b3b5fa1fcba696217ec553456d/bin/acceptance-tests.sh#L20-L28
I hope it helps.

**Reply by danielbachhuber** [reply]
@szepeviktor Thanks for the suggestion!

It would be nice if we could check this directly in the workflow file. However, I suppose it's fine to check this in our script. We added this in https://github.com/wp-cli/wp-cli-tests/pull/167/files#diff-7c3b1f709a82cb16285967a4a8fce6aec08b151c46054b509a1ecb5d07d93926R50-R60

**Reply by szepeviktor** [reply]
> It would be nice if we could check this directly in the workflow file.

Yes. This while loop could move here
https://github.com/wp-cli/.github/blob/db53c6ea1f07085483d9391070cfd8f5a177d4b9/.github/workflows/reusable-testing.yml#L161
